client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonNegativeDecimalCell.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { CellProps } from '../../../columns';
+import { NonNegativeNumberInput } from '@common/components';
+import { RecordWithId } from '@common/types';
+import { useBufferState, useDebounceCallback } from '@common/hooks';
+
+// where NonNegative is n >=0
+export const NonNegativeDecimalCell = <T extends RecordWithId>({
+  rowData,
+  column,
+  rows,
+  isDisabled = false,
+}: CellProps<T>): React.ReactElement<CellProps<T>> => {
+  const [buffer, setBuffer] = useBufferState(
+    column.accessor({ rowData, rows })
+  );
+
+  const updater = useDebounceCallback(column.setter, [column.setter], 250);
+
+  return (
+    <NonNegativeNumberInput
+      disabled={isDisabled}
+      InputProps={{ sx: { '& .MuiInput-input': { textAlign: 'right' } } }}
+      type="number"
+      value={buffer}
+      onChange={newValue => {
+        const intValue = Math.round(newValue * 100) / 100;
```

Comment:
:nit it isn't an integer value now.. so the variable name of `intValue` could be misleading for people reading later who don't look at the functionality carefully